etcdutl/snapshot/v3_snapshot.go

```diff
@@ -110,6 +114,30 @@ type Status struct {
 	Version string `json:"version"`
 }
 
+type HashKV struct {
+	Hash            uint32 `json:"hash"`
+	HashRevision    int64  `json:"hashRevision"`
+	CompactRevision int64  `json:"compactRevision"`
```

Comment:
I think `CompactRevision` in the response is noise for end users. I am inclined to remove it just like `etcdctl endpoints hashkv` https://github.com/etcd-io/etcd/blob/main/etcdctl/README.md#endpoint-hashkv

Reply:
@chaochn47 are you suggesting to remove `CompactRevison` from simple and table outputs but keep in JSON output?

Reply:
Yeah, that matches with my expectation.

Reply:
I'm against removing `CompactRevision` it is important part of confirming data corruption and hiding this value would lead to false positives.

Reply:
> I'm against removing CompactRevision it is important part of confirming data corruption and hiding this value would lead to false positives.


Agreed. I am even thinking about display both `ScheduledCompactKeyName` and `FinishedCompactKeyName`.  If the latest compaction somehow did not complete, we may get different hashes for different files even they have the same `CompactRevision`  value.  FYI. https://github.com/etcd-io/etcd/issues/15919

Reply:
> I'm against removing CompactRevision it is important part of confirming data corruption and hiding this value would lead to false positives.

It's fair point. But as Benjamin mentioned, compact is an asynchronous operation which does not guarantee `CompactRevision` indicating any revision before it is deleted. 

How about wait `FinishedCompactKeyName` value is consistent with `ScheduledCompactKeyName` after `mvcc.NewStore()`?

Reply:
> How about wait FinishedCompactKeyName value is consistent with ScheduledCompactKeyName value to guarantee the previous scheduledCompaction is completed after mvcc.NewStore()?


The most useful case to run `etcdutl hashkv` is that etcd instance isn't running anymore. `wait` makes no sense.

Reply:
> The most useful case to run etcdutl hashkv is that etcd instance isn't running anymore.

Isn't `mvcc.NewStore()` called in `etcdutl hashkv` will resume the scheduled compaction if any?  https://github.com/etcd-io/etcd/blob/1200b1006da69f555ac1959e70bdddf2f1cc9226/server/mvcc/kvstore.go#L396

If we decide to expose compactRevision, sounds like this compaction must be finished.

Reply:
It isn't good for an offline tool (`etcdutl` in this case) to silently modify the db file. But since it's doing the right thing (compaction), so I am less diligent to against it.

There is race condition in current implementation, the result is uncertain if previous compaction did not finish. If the compacted keys is bigger than batchNum, it will commit the transaction multiple times, then the hashkv may get an intermediate result. https://github.com/etcd-io/etcd/blob/8da2a5bf46cf963373d2841ed90e33560865884c/server/storage/mvcc/kvstore_compaction.go#L83

So proposal:
- Do not silently modify the data. Instead, if `etcdutl` detects any unfinished compaction, then stop the operation, and notify user, and let users to explicitly do it. 
- Add two commands, one for finishing previous unfinished compaction; the other is for calculating the hashkv of the db file (it's what this PR is doing).

Reply:
> It isn't good for an offline tool (etcdutl in this case) to silently modify the db file.
Oh you are right but anyway it is silently modify the db file in `mvcc.NewStore()` ..

Fail the command explicitly option looks good to me.